### PR TITLE
Adds `Head` and `Body` methods

### LINKED ISSUE #1: New properties for body and header on fluentRequest structure
By now we are only able to do simple requests without a Body or a Header. To support new methods like POST or PUT we will need this base

### PR DESCRIPTION
**Pull request by ArmoryNode**
Adds `Head` and `Body` methods to `fluentRequest` and adds tests for POST, PUT, and PATCH.

Fixes #1

### CODE REVIEW

File: fluentrequest.go [hunk 1]
```diff
@@ -28,8 +31,25 @@ func (r *fluentRequest) Method(method string) *fluentRequest {
...
+}
+
 func (r *fluentRequest) Run() (*http.Response, error) {
 
-	req, _ := http.NewRequest(r.method, r.url, nil)
+	req, _ := http.NewRequest(r.method, r.url, r.body)
+
+	if r.header != nil {
+		req.Header = r.header
+	}
```
Comment (VitorEstevam) [inline fluentrequest.go:52]: Is this nil verification really necessary? I did some tests and r.header had no problems in being nil

Comment (vitorestevam) [inline fluentrequest.go]: Is this nil verification really necessary? I did some tests and r.header had no problems in being nil

**Review by vitorestevam — COMMENTED** [review]
Thanks for the work here! I have a single question about a nil verification.

File: fluentrequest.go (hunk 1 above)
Comment (ArmoryNode) [reply fluentrequest.go]: Looks like you're right. I'll push the changes 👍🏼

**Pull request merged**